### PR TITLE
backport: bitcoin#17938, #19083, #19177, #19233, #19260, #19276, #19282, #19403, #19252, bitcoin-core/gui#3, #6

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Just regular bitcoin backports, mostly from v21

## What was done?
 - bitcoin/bitcoin#19233
 - bitcoin/bitcoin#19177
 - bitcoin/bitcoin#19282
 - bitcoin/bitcoin#19276
 - bitcoin-core/gui#3
 - bitcoin/bitcoin#17938
 - bitcoin-core/gui#6
 - bitcoin/bitcoin#19403
 - bitcoin/bitcoin#19083
 - bitcoin/bitcoin#19260
 - bitcoin/bitcoin#19252
 
## How Has This Been Tested?
Run unit/functional tests

Note for reviewers:
 - bitcoin#17938 has been done before except couple tiny changes which are done now
 - bitcoin#19083 is partial due to one test that fails. See a commit message for details

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
LGTM; but tests fail. Please fix :)

**Review by PastaPastaPasta — DISMISSED** [review]
force pushes seem like correct diffs to fix issues

utACK for merge via merge commit

**Comment by knst** [comment]
force push to fix test: https://github.com/dashpay/dash/compare/c7372a970caa4aceb225831bcacf8e2891ed01b0..c4f9250f75b19ac38a512d78af207811397e8031
(and make one of backport partial, 2nd force-push is trivial)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/p2p_filter.py [hunk 1]
```diff
@@ -63,107 +90,147 @@ def set_test_params(self):
...
-
-        self.log.info('Check that too large filter is rejected')
-        with self.nodes[0].assert_debug_log(['Misbehaving']):
-            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS+1))
-        with self.nodes[0].assert_debug_log(['Misbehaving']):
-            filter_node.send_and_ping(msg_filterload(data=b'\xbb'*(MAX_BLOOM_FILTER_SIZE+1)))
-
-        self.log.info('Check that too large data element to add to the filter is rejected')
-        with self.nodes[0].assert_debug_log(['Misbehaving']):
-            filter_node.send_and_ping(msg_filteradd(data=b'\xcc'*(MAX_SCRIPT_ELEMENT_SIZE+1)))
```
[the remark this answers is not in the document]
Comment (knst) [reply test/functional/p2p_filter.py]: these lines supposed to be dropped due to merge mistake during https://github.com/bitcoin/bitcoin/pull/18726/files these tests have been twice:
 - first when test_size_limit() is called https://github.com/dashpay/dash/pull/5808/commits/9a984f8a927f35a1813fba5e788fa67a83ec1167#diff-c5c320cd909288d7cf2d82632c6bafcb9085413bfddf5edf361f288dbd76e0cbL98
 - second time is right here (removed lines)

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/p2p_filter.py [hunk 2]
```diff
@@ -63,106 +90,146 @@ def set_test_params(self):
...
+        # txid = ... unused due to see TODO below
+        self.nodes[0].sendtoaddress(filter_address, 90)
+
+        self.log.debug("Send a mempool msg after connecting and check that the tx is received")
+        self.nodes[0].add_p2p_connection(filter_peer)
+        filter_peer.send_and_ping(filter_peer.watch_filter_init)
+        self.nodes[0].p2p.send_message(msg_mempool())
+        # TODO: somehow in our imlementation the last message on peer is not tx `txid`
+        # but somehow it's a "NOT FOUND" message.
+        # filter_peer.wait_for_tx(txid)
```
[the remark this answers is not in the document]
Comment (knst) [reply test/functional/p2p_filter.py:137]: but with your patch other test fails:

```
5/6 - p2p_nobloomfilter_messages.py failed, Duration: 69 s

stdout:
2024-01-16T13:21:42.140000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_∋_🏃_20240116_202140/p2p_nobloomfilter_messages_3
2024-01-16T13:21:42.652000Z TestFramework (INFO): Test that peer is disconnected if it sends mempool message
2024-01-16T13:21:44.654000Z TestFramework (INFO): Test that peer is disconnected if it sends filterload message
2024-01-16T13:21:46.657000Z TestFramework (INFO): Test that peer is disconnected if it sends filteradd message
2024-01-16T13:21:48.660000Z TestFramework (INFO): Test that peer is disconnected if it sends a filterclear message
2024-01-16T13:22:49.686000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: ''''
        test_function = lambda: not self.is_connected
'''
2024-01-16T13:22:49.687000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/knst/projects/dash/test/functional/test_framework/test_framework.py", line 155, in main
    self.run_test()
  File "/home/knst/projects/dash/test/functional/p2p_nobloomfilter_messages.py", line 44, in run_test
    self.test_message_causes_disconnect(msg_filterclear())
  File "/home/knst/projects/dash/test/functional/p2p_nobloomfilter_messages.py", line 30, in test_message_causes_disconnect
    peer.wait_for_disconnect()
  File "/home/knst/projects/dash/test/functional/test_framework/mininode.py", line 454, in wait_for_disconnect
    self.wait_until(test_function, timeout=timeout)
  File "/home/knst/projects/dash/test/functional/test_framework/mininode.py", line 450, in wait_until
    wait_until(test_function, timeout=timeout, lock=mininode_lock, timeout_factor=self.timeout_factor)
  File "/home/knst/projects/dash/test/functional/test_framework/util.py", line 243, in wait_until
    raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
AssertionError: Predicate ''''
        test_function = lambda: not self.is_connected
''' not true after 60.0 seconds
```

hm... let me think

[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/p2p_filter.py]: 👀  I don't see any issues in my branch (this PR merged on top of develop d0a6284817 + 2 commits I mentioned earlier), neither locally nor in CI https://gitlab.com/UdjinM6/dash/-/jobs/5934477704

[the remark this answers is not in the document]
Comment (knst) [reply test/functional/p2p_filter.py]: strange, CI on github succeed: https://gitlab.com/dashpay/dash/-/pipelines/1140283342
your branch on my localhost succeed
my branch succeed on localhost only after full clean & build.

probably some build glitch.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
re-utACK

**Pull request merged**